### DISCUSSION TITLE
Insert image tag instead of base64?

### DISCUSSION THREAD

**Discussion by tamerkalieh** [ueberdosis/tiptap#933] [General Questions]
Hello, when I'm adding an image to the editor, it is converted to base64. I'm saving the HTML directly in the Database. But because of the Base64, the HTML String is thousands of characters long. So I would like to add some "custom" HTML. 

Is it possible, that by clicking on a custom button, that I add the image to the editor body and under the hood, the HTML is something like: `<img src="link/to/image" />` to the editors content? Or something like that? Goal is, that I can get the `<img src="xxx" />` Tag and save that to the database

**Comment by hanspagel** [comment]
Hm… that’s how the Image example already works, or did I miss something? We don’t do anything with base64. 

https://tiptap.dev/images

**Reply by tamerkalieh** [reply]
I've just added an image to the tiptap side. Once it is on the page and I inspect the source code, I see also on the tiptap.dev page the base64. 

![Bildschirmfoto 2021-01-15 um 17 10 05](https://user-images.githubusercontent.com/17314534/104750561-91357080-5754-11eb-991e-e3bfe030cd50.png)


> Hm… that’s how the Image example already works, or did I miss something? We don’t do anything with base64.
> 
> https://tiptap.dev/images

**Reply by andreasvirkus** [reply]
> We don't do anything with base64.

@hanspagel code seems to differ with you 😛  - https://github.com/ueberdosis/tiptap/blob/main/packages/tiptap-extensions/src/nodes/Image.js#L108

https://developer.mozilla.org/en-us/docs/Web/API/FileReader/readAsDataURL

**Reply by duhaime** [reply]
tiptap 2.0 seems to have removed the `readAsDataURL` call
